### PR TITLE
Move project.json dependency completion & hover to vscode-omnisharp

### PR DESCRIPTION
**Pull request by DustinCampbell**
This change adds the project.json completion and hover support to C# for Visual Studio Code, allowing it to be removed from VS Code.

This PR supersedes https://github.com/OmniSharp/omnisharp-vscode/pull/224.

cc @aeschli

### CODE REVIEW

**Comment by DustinCampbell** [comment]
Hey @aeschli: Both jsonc-parser and request-light appear to be designed for VS Code extension authors. Is there a list somewhere of packages intended for VS Code extensions?

File: package.json [hunk 1]
```diff
@@ -33,9 +33,11 @@
     "fs-extra": "^1.0.0",
     "http-proxy-agent": "^1.0.0",
     "https-proxy-agent": "^1.0.0",
+    "jsonc-parser": "^0.3.0",
```
Comment (gregg-miskelly) [inline package.json]: Do we need to update our ThirdPartyNotices.txt for this?

Comment (DustinCampbell) [inline package.json]: I don't *think* so. That's actually produced by the VS Code team and published by @aeschli [here](https://www.npmjs.com/package/jsonc-parser).

**Comment by DustinCampbell** [comment]
Any comments here @aeschli?

**Comment by aeschli** [comment]
@DustinCampbell Good point, we don't have such a list. I created https://github.com/Microsoft/vscode-docs/issues/846

**Comment by aeschli** [comment]
jsonc-parser and request-light are both owned by Microsoft. No need to list in the ThirdPartyNotices.

**Comment by DustinCampbell** [comment]
@gregg-miskelly: Any other feedback? If not, I'm going to go ahead and take this so that @aeschli can remove the code on the VS Code side for their March release.

@aeschli: Currently, I find that this causes duplicate hover tooltips due to having HoverProviders in both VS Code and C# for VS Code. However, the experience isn't *too* terrible, and it's a point in time problem.

File: src/features/json/projectJSONContribution.ts [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+        }
+        return null;
+    }
+
+    public collectDefaultSuggestions(resource: string, result: ISuggestionsCollector): Thenable<any> {
+        let defaultValue = {
+            'version': '{{1.0.0-*}}',
+            'dependencies': {},
+            'frameworks': {
+                'dnx451': {},
```
Comment (gregg-miskelly) [inline src/features/json/projectJSONContribution.ts]: Should we update these to something more modern?

Comment (DustinCampbell) [inline src/features/json/projectJSONContribution.ts]: Yes. We can do it later though. These will still work for now.

**Comment by DustinCampbell** [comment]
@aeschli: Could you cc me on the corresponding change in VS Code?

**Comment by aeschli** [comment]
@DustinCampbell I removed the project.json support from VSCode. This will be part of the February release (in about 2 weeks)

**Comment by DustinCampbell** [comment]
Got it. Thanks!

**Comment by DustinCampbell** [comment]
Also, @aeschli, sorry that took so long!

**Pull request merged**